vm/actor/src/builtin/verifreg/mod.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+mod state;
+mod types;
+pub use self::state::State;
+pub use self::types::*;
+use crate::builtin::singletons::STORAGE_MARKET_ACTOR_ADDR;
+use crate::{HAMT_BIT_WIDTH, SYSTEM_ACTOR_ADDR};
+use address::Address;
+use ipld_blockstore::BlockStore;
+use ipld_hamt::Hamt;
+use message::Message;
+use num_bigint::ToBigUint;
+use num_derive::FromPrimitive;
+use num_traits::Zero;
+use runtime::Runtime;
+use vm::{ActorError, ExitCode, METHOD_CONSTRUCTOR};
+/// Account actor methods available
+#[derive(FromPrimitive)]
+#[repr(u64)]
+pub enum Method {
+    Constructor = METHOD_CONSTRUCTOR,
+    AddVerifier = 2,
+    RemoveVerifier = 3,
+    AddVerifiedClient = 4,
+    UseBytes = 5,
+    RestoreBytes = 6,
+}
+
+pub struct Actor;
+impl Actor {
+    /// Constructor for Registry Actor
+    pub fn constructor<BS, RT>(rt: &mut RT, root_key: Address) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_is(std::iter::once(&*SYSTEM_ACTOR_ADDR))?;
+        let empty_root = Hamt::<String, _>::new_with_bit_width(rt.store(), HAMT_BIT_WIDTH)
+            .flush()
+            .map_err(|e| {
+                ActorError::new(
+                    ExitCode::ErrIllegalState,
+                    format!("Failed to create registry state {:}", e),
+                )
+            })?;
+        let st = State::new(empty_root, root_key);
+        rt.create(&st)?;
+        Ok(())
+    }
+
+    pub fn add_verifier<BS, RT>(rt: &mut RT, params: AddVerifierParams) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        let state: State = rt.state()?;
+        rt.validate_immediate_caller_is(std::iter::once(&state.root_key))?;
+
+        rt.transaction::<_, Result<_, ActorError>, _>(|st: &mut State, rt| {
+            st.put_verifier(rt.store(), &params.address, &params.allowance)
+                .map_err(|e| {
+                    ActorError::new(
+                        ExitCode::ErrIllegalState,
+                        format!("failed to add verifier: {:}", e),
+                    )
+                })?;
+            Ok(())
+        })??;
+
+        Ok(())
+    }
+
+    pub fn delete_verifier<BS, RT>(rt: &mut RT, params: AddVerifierParams) -> Result<(), ActorError>
```

Comment:
Best to just keep naming consistent (confusing to have a method name that differs 
```suggestion
    pub fn remove_verifier<BS, RT>(rt: &mut RT, params: AddVerifierParams) -> Result<(), ActorError>
```